### DISCUSSION TITLE
Préremplissage - Indiquer si un champ et optionnel ou obligatoire

### DISCUSSION THREAD

**Discussion by ChrisThe01** [MTESSDev/FRW#154] [Annonces]
Peut-on faire du pré remplissage pour indiquer si un champ est obligatoire ou non ?

**Comment by j0ber** [comment]
Bonjour,
Il faudrait en savoir plus sur ce que vous voulez faire exactement, mais si c'est uniquement d'avoir un champ obligatoire ou facultatif en fonction d'un indicateur ou de la valeur d'un autre champ c'est possible.
Le pré-remplissage n'affecte pas les validations, mais pourrait spécifier un indicateur qui lui est utilisé pour déterminer si le champ est obligatoire ou non.

Avez-vous le cas concret, histoire qu'on comprenne un peu plus?

Merci

**Reply by ChrisThe01** [reply]
Par exemple,

Dans le formulaire, nous avons des sections d’où dépendamment d’un indicateur, le champ sera affiché avec un étoile ou non et il y aura une validation ou non. Mais c’est le même champ. Une pièce joint par exemple.

**Reply by j0ber** [reply]
D'accord, pour faire cela il vous suffit de mettre le champ 2 fois dans le formulaire (un à la suite de l'autre, très important) et chacun doit avoir un "v-if".

Le premier champ aurait un "v-if", disons v-if: maCondition === true, et le second v-if: maCondition === false.
De cette façon un seul des 2 champs sera affiché en tout temps.
Pour chaque champ tu peux avoir une valeur différente pour "validations". 

Par défaut les champs sont requis. Donc si c'est ta 2ème condition qui doit afficher le champ mais optionnel, il suffira de mettre :
```YAML
validations:
    optional: 
```
dans la configuration du second champ.

**Reply by ChrisThe01** [reply]
Est-ce que le le même nom de champ peut être utilisé ?

**Reply by j0ber** [reply]
Il FAUT que ce soit le même nom de champ. Et comme mentionné précédemment, les 2 doivent se suivre dans la config. Très important.

**Reply by ChrisThe01** [reply]
Les 2 doivent se suivre dans la config.  ? Désolé si je ne comprends pas.

**Reply by j0ber** [reply]
Dans le fichier de config, la déclaration des champs doit se suivre... Ex. Il ne peut pas y avoir un autre champ entre les 2.

**Reply by ChrisThe01** [reply]
Oui je comprend, mais c'est le MÊME nom de champ.

**Reply by j0ber** [reply]
OUI! "name" doit être identique. Un seul des deux sera affiché (votre condition v-if doit s'assurer que les deux champs ne doivent pas être visibles en même temps).

**Reply by ChrisThe01** [reply]
Merci
